### PR TITLE
webview: simple GRPC server

### PR DESCRIPTION
**Pull request by jazzdan**
The point of this commit is just to validate the skeleton of the GRPC server,
not to have something useful.

This will be expanded in the future to fully implement all of the APIs
in the HUD webserver (analytics, triggers, snapshots, etc) EXCEPT the
websocket.

Other future work includes generating a TypeScript client for use
in the web frontend.

### CODE REVIEW

File: pkg/webview/view.proto [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  string tiltCloudUsername = 8;
+  string tiltCloudSchemeHost = 9;
+  string tiltCloudTeamID = 10;
+
+  string fatalError = 11;
+}
+
+message GetViewRequest {}
+
+service ViewService {
```
Comment (jazzdan) [inline pkg/webview/view.proto]: Is this the right name for this service?

File: Makefile [hunk 2]
```diff
@@ -24,7 +24,7 @@ scripts/protocc/protocc.py: scripts/protocc
 	git submodule update
 
 proto: scripts/protocc/protocc.py
-	python3 scripts/protocc/protocc.py --out go
+	python3 scripts/protocc/protocc.py --out go && toast proto
```
Comment (nicks) [inline Makefile]: this isn't right. right now, this makefile rule only depends on scripts/protocc/protocc.py.

can you create a new makefile rule with the correct dependencies?

File: internal/hud/server/server.go [hunk 3]
```diff
@@ -64,11 +68,19 @@ func ProvideHeadsUpServer(
...
 		store:    store,
 		router:   r,
 		a:        analytics,
 		uploader: uploader,
+		grpcMux:  grpcMux,
+	}
+
+	err := proto_webview.RegisterViewServiceHandlerServer(context.TODO(), grpcMux, s)
+	if err != nil {
+		// TODO(dmiller) what should we do here?
```
Comment (nicks) [inline internal/hud/server/server.go]: let's change this provider to return an error and re-run `wire`. it should handle this correctly.

File: internal/hud/server/server.go [hunk 4]
```diff
@@ -64,11 +68,19 @@ func ProvideHeadsUpServer(
...
+	grpcMux := runtime.NewServeMux()
 	s := &HeadsUpServer{
 		store:    store,
 		router:   r,
 		a:        analytics,
 		uploader: uploader,
+		grpcMux:  grpcMux,
+	}
+
+	err := proto_webview.RegisterViewServiceHandlerServer(context.TODO(), grpcMux, s)
```
Comment (nicks) [inline internal/hud/server/server.go]: let's change this provider to take a context.Context and re-run `wire`. It should handle this correctly.

File: toast.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  apk_update:
+    command: apk update
+  install_protoc_deps:
+    dependencies:
+      - apk_update
+    command: apk add protobuf protobuf-dev git
+  install_protoc_gen_go:
+    dependencies:
+      - install_protoc_deps
+    command: go get -u github.com/golang/protobuf/protoc-gen-go github.com/golang/protobuf/jsonpb github.com/golang/protobuf/proto github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway google.golang.org/grpc
```
Comment (nicks) [inline toast.yml]: i'm a little bit nervous about these not being vendored properly, and that causing a lot of problems. protocc was meant to solve these problems

File: Makefile [hunk 6]
```diff
@@ -170,3 +173,9 @@ prettier:
 storybook:
 	cd web && yarn install
 	cd web && yarn storybook
+
+tilt-container:
```
Comment (nicks) [inline Makefile]: maybe call this tilt-toast-container?

this should also be in the .PHONY declaration

File: internal/hud/server/server.go [hunk 7]
```diff
@@ -56,19 +59,28 @@ type HeadsUpServer struct {
...
+	grpcMux           *runtime.ServeMux
 }
 
 func ProvideHeadsUpServer(
 	store *store.Store,
 	assetServer assets.Server,
 	analytics *tiltanalytics.TiltAnalytics,
-	uploader cloud.SnapshotUploader) *HeadsUpServer {
+	uploader cloud.SnapshotUploader,
+	ctx context.Context) (*HeadsUpServer, error) {
```
Comment (nicks) [inline internal/hud/server/server.go]: fwiw, i would usually put ctx first, just as a style thing

File: pkg/webview/view.proto (hunk 1 above)
Comment (nicks) [reply pkg/webview/view.proto]: ya seems fine

File: pkg/webview/view.proto [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+package webview;
+
+import "google/api/annotations.proto";
+
+// TODO(dmiller) fill this in
+message Resource {}
+
+message TiltBuild {
+  string version = 1;
+  string commitSHA = 2;
```
Comment (nicks) [inline pkg/webview/view.proto]: proto naming style is slug case - commit_SHA

File: pkg/webview/view.proto [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  string date = 3;
+  bool dev = 4;
+}
+
+message View {
+  string log = 1;
+  repeated Resource resources = 2;
+  bool logTimestamps = 3;
```
Comment (nicks) [inline pkg/webview/view.proto]: ditto here - log_timestamps

File: pkg/webview/view.proto [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+  map<string, bool> featureFlags = 4;
```
Comment (nicks) [inline pkg/webview/view.proto]: and so on

**Pull request merged**